src/mqtt/client.rs

```diff
@@ -73,12 +73,12 @@ pub struct MqttClientConfiguration<'a> {
     pub skip_cert_common_name_check: bool,
     #[cfg(not(esp_idf_version = "4.3"))]
     pub crt_bundle_attach: Option<unsafe extern "C" fn(conf: *mut c_types::c_void) -> esp_err_t>,
+    
+    pub cert_pem: &'a [u8],
```

Comment:
The `'a` lifetime would only work if MQTT is making a copy of the certificates prior to starting. You can check the C code to confirm deny this. However:
- Given how memory constrained the esp32 is
- Given that ESP IDF HTTP [client](https://github.com/esp-rs/esp-idf-svc/blob/master/src/http/client.rs#L71) and [server](https://github.com/esp-rs/esp-idf-svc/blob/master/src/http/server.rs#L46) do not make a copy of the certs but just use whatever the user had provided as pointers

... I would be very surprised if MQTT client makes a copy of those.
So if you can double check what MQTT C driver does, and if it indeed does not copy the certs, you need to:
- Replace `'a` with `'static`
- (for consistency with HTTP client & server) change `[u8]` to `str`

... given that most certificates are encoded into the binary code/flash itself with `include_bytes!` or `include_str!` `'static`-fying the above pointers is not a big constraint. Basically, you would:
- `const MY_CERT: &'static str = include_str!("my_cert")`
- And then just pass `MY_CERT` to the MQTT configuration

What we can do in future (but no rush for now) would be to generify the MQTT client over a `'a` lifetime, whould would allow you to use non-static certificates as well. As in heap-allocated.

Reply:
Reminder: please also change `[u8]` to `str` (and then `include_bytes` to `include_str` in your own code) and if you have some familiarity with C - check the MQTT driver C code as to how the certs are used (as-is, or copied).